src/Functions/parseDateTime.cpp

```diff
@@ -437,6 +439,16 @@ namespace
             return {};
         }
 
+        [[nodiscard]]
+        VoidOrError setMicrosecond(Int32 microsecond_)
+        {
+            if (microsecond_ < 0 || microsecond_ > 999999)
```

Comment:
999999 seems arbitrary. Users who specify more than 6 x `S` placeholders will expect nanosecond precision. I think microsecond precision is generally okay but then we should disallow more than 6 x `S` in the format string.